balancer-js/src/modules/liquidity/liquidity.module.ts

```diff
@@ -0,0 +1,132 @@
+import {
+    BigNumber,
+    parseFixed,
+    formatFixed,
+    BigNumberish,
+} from '@ethersproject/bignumber';
+import { BalancerSdkConfig, TokenBalance, TokenPrice } from '@/types';
+import { Pools } from '@/modules/pools/pools.module';
+import { SubgraphPoolBase } from '@balancer-labs/sor';
+import { TokenProvider } from '../data-providers/token/provider.interface';
+import { PoolProvider } from '../data-providers/pool/provider.interface';
+import { TokenPriceProvider } from '../data-providers/token-price/provider.interface';
+import { PoolType } from '../pools/types';
+import { Zero } from '@ethersproject/constants';
+
+const SCALING_FACTOR = 36;
+
+export interface PoolLiquidity {
+    address: string;
+    liquidity: string;
+}
+
+export class Liquidity {
+    constructor(
+        config: BalancerSdkConfig,
+        private pools: PoolProvider,
+        private tokens: TokenProvider,
+        private tokenPrices: TokenPriceProvider
+    ) {}
```

Comment:
isn't liquidity just a sum of token balance * it's price? how about just returning a token[] from this and running the calculation on it? reason is to remove a hidden dependency on a Pools class.

Reply:
We need the pools provider for recursive liquidity calculations, as it loads the subpools, loads the tokens from them, and calculates their liquidity.